### PR TITLE
Add a tag feature to the deck builder

### PR DESCRIPTION
**Pull request by alicetouhou**
Built off the deck builder upgrades branch.
Adds a tag feature to the deck builder for more easily searching through decks

Not planning on adding modifying already-created tags in this PR, but it's easily able to be added in the future.

### CODE REVIEW

**Comment by Screaper91** [comment]
I approve the functionality, needs work on UI composition and viewing tag colors in the filter dropdown for duplicate names

**Comment by alicetouhou** [comment]
Working on fixing spaghetti code, but for now suggestions for UI would be appreciated

File: client/src/app/deck/deck-select.tsx
Comment (Screaper91) [inline client/src/app/deck/deck-select.tsx:1]: The manage tags button is not included in mobile children filter group (see line 564)

File: client/public/images/hermits-emoji/teto2.png
Comment (Screaper91) [inline client/public/images/hermits-emoji/teto2.png:1]: Is this supposed to be included?

File: client/src/logic/saved-decks/saved-decks.ts [hunk 1]
```diff
@@ -94,3 +95,40 @@ export const convertLegacyDecks = (): number => {
...
+}
+
+export const saveTag = (tag: Tag) => {
+	const createdTags = getCreatedTags()
+	const hash = 'Tag_' + tag.key
+	if (createdTags.find((createdTag) => createdTag.key === tag.key)) return
+	localStorage.setItem(hash, JSON.stringify(tag))
+}
+
+export const deleteTag = (tag: Tag) => {
```
Comment (Screaper91) [inline client/src/logic/saved-decks/saved-decks.ts:131]: This does not remove references to the deleted tag in saved decks

File: common/config/server-config.json [hunk 2]
```diff
@@ -18,5 +18,5 @@
 		"maxDuplicates": 3,
 		"maxDeckCost": 42
 	},
-	"logoSubText": "Tournament Time!"
+	"logoSubText": "中國加油!"
```
Comment (Screaper91) [inline common/config/server-config.json:21]: Is this an intentional change for this PR?

**Review by Screaper91 — CHANGES_REQUESTED** [review]
The manage tags button is not included in the mobile children filter group and deleted tags are not removed from decks. There are also some changes that are not related to this PR's purpose.

File: client/public/images/hermits-emoji/teto2.png
Comment (alicetouhou) [reply client/public/images/hermits-emoji/teto2.png:1]: LOL. I totally thought I fixed that haha 😂

File: common/config/server-config.json (hunk 2 above)
Comment (alicetouhou) [reply common/config/server-config.json:21]: Nope, was just experimenting with how non latin characters are displayed

File: client/src/app/deck/deck-select.tsx
Comment (alicetouhou) [reply client/src/app/deck/deck-select.tsx:1]: Fixed

File: client/src/logic/saved-decks/saved-decks.ts (hunk 1 above)
Comment (alicetouhou) [reply client/src/logic/saved-decks/saved-decks.ts:131]: Fixed

**Review by Screaper91 — APPROVED** [review]

**Pull request merged**